From 2f9155ddd8999ce57facf7b7704030c7d14c58d5 Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Mon, 26 Nov 2018 21:56:00 +0200
Subject: [PATCH 0810/1051] net: mvpp2: use skb-cloned field but not proc in
 recycle

Do not use skb_cloned() procedure but use the field skb->cloned
for a buffer recycling decision.
The skb->cloned covers 2 cases: skb_cloned() and skb_header_cloned()
and also executed faster since there is no read-atomic.

Change-Id: I893567b9fdc0c842b09b04f1d045b4dd1933a800
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61339
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 81fdccf16b8a..0d6b7cf9ef62 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -3264,7 +3264,8 @@ static int mvpp2_recycle_get_bm_id(struct sk_buff *skb)
 	if (!MVPP2_RXTX_HASH_IS_OK(skb, hash))
 		return -1;
 	/* Check if skb could be free */
-	if (skb_shared(skb) || skb_cloned(skb))
+	/* Use skb->cloned but not skb_cloned(), skb_header_cloned() */
+	if (skb_shared(skb) || skb->cloned)
 		return -1;
 	/* WA: don't put to recycle the buffer with fully consumed headroom */
 	if (skb_headroom(skb) <= MVPP2_MH_SIZE)
-- 
2.17.1

